sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -815,6 +815,15 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  val HIVE_METASTORE_PARTITION_PRUNING_INSET_THRESHOLD =
+    buildStaticConf("spark.sql.hive.metastorePartitionPruningInSetThreshold")
+      .doc("The threshold of set size for InSet predicate when pruning partitions through Hive " +
+        "Metastore. Larger values may cause Hive metastore stack overflow.")
+      .version("3.1.0")
+      .intConf
+      .checkValue(_ > 0, "The value of metastorePartitionPruningInSetThreshold must be positive")
+      .createWithDefault(Int.MaxValue)
```

Comment:
shall we give a reasonable default value?

Reply:
Set it to 1000 because the default value of `hive.exec.max.dynamic.partitions` is also 1000.